lib/nfg/capistrano/tasks/deploy.rake

```diff
@@ -0,0 +1,37 @@
+require 'aws-sdk-s3'
+
+namespace :deploy do
+  Rake::Task['deploy:compile_assets'].clear
+
+  desc 'Download or compile assets'
+  task :compile_assets => [:set_rails_env] do
+    assets_cache_prefix = Nfg::Capistrano::Config.circleci[:parameters][:assets_cache_prefix][:default]
+    revision = %x(git --no-pager log -1 --pretty=format:%H)
+    branch = fetch(:branch).gsub('/', '_')
+    assets_filename="#{assets_cache_prefix}-assets-#{revision}-#{branch}.tar.gz"
+    s3_bucket = Nfg::Capistrano::Config.circleci[:parameters][:s3_bucket][:default]
+    aws_credentials = Aws::Credentials.new(
+      Nfg::Capistrano::Config.api_keys[:aws_access_key_id],
+      Nfg::Capistrano::Config.api_keys[:aws_secret_access_key]
+    )
+    s3 = Aws::S3::Client.new(
+      region: ENV.fetch('AWS_REGION', 'us-east-1'),
+      credentials: aws_credentials
+    )
+
+    begin
+      s3.head_object(bucket: s3_bucket.gsub('s3://', ''), key: "assets/#{assets_filename}")
+      on release_roles :all do
+        execute :s3cmd, "--force get #{s3_bucket}/assets/#{assets_filename} #{shared_path}/public/assets/#{assets_filename}"
+        info Airbrussh::Colors.green("Downloaded #{assets_filename} from #{s3_bucket}/assets")
+        execute "tar zxvf #{shared_path}/public/assets/#{assets_filename} -C #{shared_path}"
+      end
+    rescue Aws::S3::Errors::NotFound => e
```

Comment:
This example is labeled "Head Object" on this page

https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/ruby_s3_code_examples.html#actions

and object_exists in the samples repo so idek ¯\\\_(ツ)\_\/¯ 

https://github.com/awsdocs/aws-doc-sdk-examples/blob/main/ruby/example_code/s3/object_exists.rb#L8

I'm far less concerned now and defer to your level of concern here on if we might be building on connection issues.

Reply:
I agree that `exists?` feels cleaner. It means I have to use `Aws::S3::Object` instead of `Client` but that's not a big deal.

I guess the only issue is whether we want to care about things like permissions errors. If so, then it seems like it would be better to retain the rescue block but modify it to capture any exceptions.

Reply:
I did refactor this to use `exists?`, so gaze upon that and see if it sparks joy.

Reply:
It looks good. We appear to still be making a head request after we confirmed it exists which is probably duplicative?

I agree we may still want the rescue block but for true exceptions.